### ISSUE TITLE
Implement “Connect GA” metric tile CTA after Key Metrics widget setup

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6263]
## Feature Description

If one or two selected metrics within the Key Metrics widget area are from GA4, but GA4 is disconnected, then a new CTA tile should be rendered to encourage the user to reconnect the module.

Refer to the [relevant section](https://docs.google.com/document/d/1k8NOo2xEHUQAcgYG0TB5sUjGcPiC8SWIalEdd9G8-9g/edit#heading=h.hfwitp2lb9g3) in the design doc for additional context.

![Screenshot 2022-12-14 at 11 31 26](https://user-images.githubusercontent.com/1721319/207584194-b916b2c2-2795-4845-b6a7-47aa8245cfa4.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A new Connect GA CTA tile should be implemented as per the [Figma mock](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=2019%3A65231&t=bwOtTQ7S3Odzwt7u-0). Note that the width of the tile should be the same as a widget's standard "quarter" width and not 1/3rd as in the Figma design. 
* The tile should be rendered if either 1 or 2 of the selected metrics are dependant on GA4 and GA4 was disconnected after the widget's setup.
* Clicking on the link in the CTA should take the user to the Connect services page for GA.

## Implementation Brief

* Wait for #6209, #6257, and #6313 are merged.
* Create a new `ConnectGA4TileCTA` component in the same directory #6209 is implemented.
* The component should render the following:
  * Use `assets/svg/graphics/analytics.svg` as the image.
  * Render the description text within a `p` tag per the Figma design. The text should be translated.
  * Render the link text within the `Link` component per the Figma design. The text should be translated. The `onClick` callback should follow a similar implementation as the `SetupModule` component:
  https://github.com/google/site-kit-wp/blob/43c436489a48caeead4c33809187b6823fb0aaaf/assets/js/components/settings/SetupModule.js#L57-L82
* The component should render `null` if the GA4 module is connected using the `isModuleConnected` selector.
* Using the `getKeyMetrics` selector, the component should check if any of the selected metrics are from GA4. If so, it should render the `CTA` component. Otherwise, it should render `null`.
* Create a new `ConnectGA4TileCTAWidget` component in the same directory #6209 is implemented.
* It takes the `Widget` and `WidgetNull` components as props.
* Render the `ConnectGA4TileCTA` component within the `Widget` component if the following conditions are met:
  * If the GA4 module isn't connected. Check using the `isModuleConnected` selector.
  * If the selected metrics are dependent on GA4. Check using the `getKeyMetrics` selector.
* Otherwise, render the `WidgetNull` component.
* Create stories for the `ConnectGA4TileCTAWidget` component.

### Register the widget

* Register the `ConnectGA4TileCTAWidget` component in the `KeyMetrics` widget area once #6313 is merged.
* The widget should be registered with the following settings:
  * `width` should be `QUARTER`.
  * `wrapWidget` should be `false`.
  * `priority` should be `1`.
  * `modules` should be `['analytics']`.

### Test Coverage

* Add tests/storybook coverage for the `ConnectGA4TileCTA` component.

## QA Brief

* Ensure that the `ga4Reporting` and `userInput` feature flags are enabled. 
* Make sure Analytics(4) module is connected. 
* Setup Key Metrics manually with the following snippet. 
```js
(async () => {
await googlesitekit.data.dispatch('core/user').setKeyMetricsSetting( 'widgetSlugs', ['kmAnalyticsTopTrafficSource', 'kmAnalyticsLoyalVisitors'] );
await googlesitekit.data.dispatch('core/user').saveKeyMetricsSettings();
})();

```
* Disconnect Analytics module.
* See the new widget appear in the Key Metrics area. 
* All other conditions (No active analytics widget, more than 2 widgets, combination of search-console and analytics)  can be tested by changing the `widgetSlugs` in the above snippet. ie. 
```js
(async () => {
await googlesitekit.data.dispatch('core/user').setKeyMetricsSetting( 'widgetSlugs', ['kmAnalyticsTopTrafficSource', 'kmAnalyticsLoyalVisitors', "kmAnalyticsPopularProducts"] );
await googlesitekit.data.dispatch('core/user').saveKeyMetricsSettings();
})();
```

## Changelog entry

* Add the Connect GA tile to Key Metrics widget.

**Comment by tofumatt** [comment]
ACs here are good 👍🏻

**Comment by tofumatt** [comment]
Similar to #6265, just an adjustment to naming (`GA4CTATileConnect` -> `ConnectGA4TileCTA`). Added storybook coverage as well.

Otherwise looks good, IB ✅

**Comment by mohitwp** [comment]
### QA Update ❌ 

@jimmymadon 

**Issue 1** : Font color for link is not consistent with other links color on dashboard and not same as font color mentioned under Figma. 

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/35277a0f-1efc-4256-974a-6d65a3b92723)

![image](https://github.com/google/site-kit-wp/assets/94359491/eaf94a47-af31-4f59-ba2e-a8fc34874b27)

</details>

**Question** : CTA text missing 'Google' as mentioned under Figma. Did we make changes to text ?

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/b50fcc79-b7bf-400e-987b-fd9693107d2a)

</details>

**Comment by mohitwp** [comment]
Thanks, @jimmymadon ! Sorry for the confusion.

**Comment by kuasha420** [comment]
@mohitwp Thank you for sharing your observations. 

I have filled a follow up [PR](https://github.com/google/site-kit-wp/pull/7205) to fix this. See the discussion on [slack](https://10up.slack.com/archives/CBKKQEBR9/p1687771737184799) for more detail on it, specially regarding accessibility. 

1. This was a conscious decision made during the development as it is consistent with how we refer Analytics plugin wide. See [here](https://github.com/google/site-kit-wp/pull/7153#pullrequestreview-1479127815) for details. 
2. The Other tiles will be tackled in a separate issue. But this one should not be visible on a fresh site if it doesn't have 1-3 KM Widgets setup that depends on Analytics. If you are using the snippet in the QAB, it will appear because the snippet is forcefully setting up KM. In a normal circumstance, it will not appear. 

Cheers.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

@kuasha420 

- Tested on staging environment.
- CTA font color issue is fixed now.

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/62e8a813-446f-4a5a-912a-d8334d233f56)

</details>

As discussed on slack 'Connect Analytics' tile will appear when 1-3 of the KM widgets are from Analytics. I use the snippet mentioned under QAB and tried different combination of 1-3 KMA widgets but 'Connect Analytics' tile is not showing on when we are using 3 KMW. In case of 2 widgets it is showing. 


**Tested below cases-** 
- Tested when Number of KMAW is 2 - **PASS**
- Tested when Number of KMAW widget is 1 - **PASS**
- Tested combination of KMAW and SC widget - **PASS**
- Tested only SC console widget - **PASS**
- Tested when Number of KMAW is 3 - **FAIL**
- Tested when Number of KMAW is 4 - **PASS**

</details>


Uploading Recording #420.mp4…

</details>

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified 'Connect Analytics CTA' tile with Figma designs.
- Verified 'Connect Analytics CTA' tile is showing in below conditions.

  -- When Number of KMAW is 2 - **PASS**
 -- When Number of KMAW widget is 1 - **PASS**
 -- When Number of KMAW is 0 - **PASS**
 -- Combination of KMAW and SC widget - **PASS**
 --Only SC console widget - **PASS**
 -- When Number of KMAW is 3 - **PASS**
 -- When Number of KMAW is 4 - **PASS**
